spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -200,4 +200,24 @@ public String getBinder(String channelName) {
 		return bindings.get(channelName).getBinder();
 	}
 
+	/**
+	 * Return configuration properties as Map.
+	 * @return map of channel binding configuration properties.
+	 */
+	public Map<String, Object> asMapProperties() {
+		Map<String, Object> properties = new HashMap<>();
+		properties.put("instanceIndex", String.valueOf(getInstanceIndex()));
+		properties.put("instanceCount", String.valueOf(getInstanceCount()));
+		for (String name : getConsumerProperties().stringPropertyNames()) {
+			properties.put(name, getConsumerProperties());
+		}
+		for (String name : getProducerProperties().stringPropertyNames()) {
+			properties.put(name, getProducerProperties());
+		}
+		for (Map.Entry<String, BindingProperties> entry : getBindings().entrySet()) {
+			properties.put(entry.getKey(), entry.getValue());
+		}
+		return properties;
+	}
+
```

Comment:
At some point (perhaps in a future milestone) we should consider some mechanism where the user (perhaps via config server) can configure exactly what is captured in the history entries.